Description: can set uid for tcpmux
 This patch fixes CVE-2013-4342, by allowing TCPMUX to be used
 under a different user.
Origin: other, https://github.com/xinetd-org/xinetd/pull/10/files
Reviewed-By: Salvo 'LtWorf' Tomaselli <tiposchi@tiscali.it>
 .
 xinetd (1:2.3.15-2) unstable; urgency=high
 .
   * Fix CVE-2013-4342 making TCPMUX services change the uid.
     (Closes: #324678)
Author: https://github.com/octurite
Bug-Debian: http://bugs.debian.org/324678
Last-Update: 2013-10-03

--- xinetd-2.3.15.orig/xinetd/builtins.c
+++ xinetd-2.3.15/xinetd/builtins.c
@@ -617,7 +617,7 @@ static void tcpmux_handler( const struct
    if( SC_IS_INTERNAL( scp ) ) {
       SC_INTERNAL(scp, nserp);
    } else {
-      exec_server(nserp);
+      child_process(nserp);
    }
 }
 
